### PR TITLE
feat(batch): add Kafka/MSK batch processing support

### LINKED ISSUE #5362: Feature request: add msk to batch processing
### Use case

msk uses the same polling lambda invocation as sqs, dynamodbstream, and kinesis streams and would benefit from batch processing but is not available as an option in the batch processor class. 

 [This article](https://aws.amazon.com/blogs/compute/efficiently-processing-batched-data-using-parallelization-in-aws-lambda/) makes it seem like it's available and should be used as best practice but it's not for msk.

### Solution/User Experience

KafkaMskEventModel and KafkaEvent are already implemented so it should be straight forward to add I assume.

### Alternative solutions

_No response_

### Acknowledgment

- [x] This feature request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [x] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by shaked-lokits**
**Issue number:** closes #5362

## Summary

Add support for processing Kafka and MSK events with partial batch failure reporting, leveraging the new `ReportBatchItemFailures` functionality that AWS recently enabled for Kafka event sources.

### Changes

- Add `EventType.Kafka` to batch processor for unified MSK/self-managed Kafka support
- Implement `_collect_kafka_failures()` for composite identifiers (topic-partition + offset)
- Add `_get_records_from_event()` helper to flatten Kafka's nested record structure
- Add `KafkaItemIdentifier` TypedDict for type safety
- Add 14 functional tests covering sync/async processing, partial failures, and edge cases
- Update batch processing documentation with Kafka section and SAM template
- Update Kafka utility documentation with cross-reference to batch processing

### User experience

**Before:** Users had to manually implement partial batch failure handling for Kafka/MSK events, including constructing the correct response format with composite identifiers.

**After:** Users can use the familiar batch processing API:

```python
from aws_lambda_powertools.utilities.batch import BatchProcessor, EventType, process_partial_response

processor = BatchProcessor(event_type=EventType.Kafka)

def record_handler(record):
    # Process record
    return record.value

def lambda_handler(event, context):
    return process_partial_response(event=event, record_handler=record_handler, processor=processor, context=context)
```

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shaked-lokits** [comment]
Hey @ConnorKirk, I have an open ticket in code review for this for the past month and I was just asked what's the status on this one?🙏

File: aws_lambda_powertools/utilities/batch/base.py [hunk 1]
```diff
@@ -365,6 +371,21 @@ def _collect_dynamodb_failures(self):
...
+            # Both data class and Pydantic model use the same field names
+            failures.append(
+                {
+                    "itemIdentifier": {
+                        "topic-partition": f"{msg.topic}-{msg.partition}",
+                        "offset": msg.offset,
+                    },
+                },
+            )
+        return failures
```
Comment (leandrodamascena) [inline aws_lambda_powertools/utilities/batch/base.py:387]: Hey, I think there's a bug here. The key used in the `itemIdentifier` dict is `"topic-partition"`, but the [AWS docs](https://docs.aws.amazon.com/lambda/latest/dg/kafka-retry-configurations.html) show the correct key is `"partition"`.

AWS expected format:
```json
{"itemIdentifier": {"partition": "topic-0", "offset": 100}}
```

Current implementation:
```json
{"itemIdentifier": {"topic-partition": "topic-0", "offset": 100}}
```

The AWS docs explicitly say: *"Any invalid topic-partition_number or offset returned that was not present in the invoked event will be treated as failure and entire batch will be retried."* - so using the wrong key name would cause Lambda to treat every response as invalid and retry the entire batch every time.

This needs to be fixed in both `_collect_kafka_failures` in `base.py` and `KafkaItemIdentifier` in `types.py`.

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
Hey @shaked-lokits thanks a lot for this PR and sorry for the delay in reviewing it, we had some internal prioritizations.

Overall the code is really nice, there is only one critical bug that must be fixed before we merge it. I left a comment.

**Review by leandrodamascena — APPROVED** [review]
Hi @shaked-lokits, I just fixed the last bug and it's ready to be merged! Thank you so much for working on it and contributing to Powertools.

APPROVED

**Review by hjgraca — APPROVED** [review]
